### ISSUE TITLE
Calling VRDisplay.requestPresent causes autoplaying video elements to stop playing on iOS devices

### ISSUE THREAD

**Issue by capnmidnight** [googlevr/webvr-polyfill#169]
I have a scene where I have a mesh textured with a video element. The video is a small fireplace and it's meant to loop indefinitely, sitting as a minor detail in a room.

This issue does not affect Android.
## Steps to Reproduce
1. Acquire a small H.264/MP4 video to use as a texture named `testVideo.mp4`.
2. Create a page `test.html`

``` html
<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
  <script src="webvr-polyfill.js"></script>
  <script src="three.js"></script>
</head>
<body>
  <script src="app.js"></script>
</body>
</html>
```
1. Create a script file `app.js`

``` javascript
var renderer = new THREE.WebGLRenderer(),
camera = new THREE.PerspectiveCamera(75, 1, 0.01, 100),
scene = new THREE.Scene(),
video = document.createElement("video"),
isiOS = /iP(hone|od|ad)/.test(navigator.userAgent || "");

document.body.appendChild(renderer.domElement);

renderer.domElement.style.position = "absolute";
renderer.domElement.style.left = 0;
renderer.domElement.style.top = 0;
renderer.domElement.style.width = "100%";
renderer.domElement.style.height = "100%";

function resize(){
  renderer.setSize(screen.width, screen.height);
}

window.addEventListener("resize", resize);
resize();

video.src = "videos/fireplace.mp4";
video.preload = "auto";
video.autoplay = true;
video.loop = true;
video.muted = true;
video.controls = true;
video.setAttribute("playsinline", "");
video.setAttribute("webkit-playsinline", "");
if(isiOS) {
  video.onend = function() {
    video.play();
  };
}

document.body.insertBefore(video, document.body.children[0]);
video.play();

var videoTexture = null,
lastTime = null;

video.oncanplay = function() {
  videoTexture = new THREE.Texture(video);
  var material = new THREE.MeshBasicMaterial({ map: videoTexture }),
  aspectRatio = video.videoHeight / video.videoWidth,
  geometry = new THREE.PlaneBufferGeometry(1, aspectRatio),
  mesh = new THREE.Mesh(geometry, material);
  mesh.position.set(0, 0, -1);
  scene.add(mesh);
};

function paint(t) {
  requestAnimationFrame(paint);
  if(videoTexture && videoTexture.image.currentTime !== lastTime) {
    lastTime = videoTexture.image.currentTime;
    videoTexture.needsUpdate = true;
  }
  renderer.render(scene, camera);
}

requestAnimationFrame(paint);

var buttonContainer = document.createElement("div");
buttonContainer.style.position = "absolute";
buttonContainer.style.top = 0;
buttonContainer.style.left = 0;
document.body.appendChild(buttonContainer);

navigator.getVRDisplays().then(function(displays) {
  console.log(displays);
  displays.forEach(function(display) {
    var button = document.createElement("button");
    button.type = "button";
    button.appendChild(document.createTextNode(display.displayName));
      button.onclick = function() {
        display.requestPresent([{source: renderer.domElement}]);
      };
      buttonContainer.appendChild(button);
    });
  }).catch((exp) => console.error(exp));
```
1. Run the page from a live web server (iOS sometimes will refuse to load a video from local dev servers).
2. Note that the video is playing and loops at on its own.
3. Click the button for entering Google Cardboard mode.
4. Note that the video stops playing.
## Problem

WebVR-Polyfill includes a WakeLock to prevent the screen from sleeping when using Google Cardboard mode, as gaze-based interfaces don't have any physical user interaction with the screen. Something about how the WakeLock works makes the video stop. In cases where there is already an autoplaying video in the page, the WakeLock is not necessary.
## Work around

Fork WebVR-Polyfill and rip out the WakeLock code.
## System Info
- webvr-polyfill v0.9.3
- Safarai on iOS v10.0

**Comment by jsantell** [comment]
Thanks for the thorough steps to reproduce! Sorry it's taken so long to respond.

The wakelock code is a hack around mobile browsers to begin with, but looks like you found a solution -- I think in the case where video is playing, and the wakelock not necessary, we can disable it, but would be difficult if not impossible to determine if the wakelock is necessary on the current browser. I think an option to optionally disable wakelock when it's not needed for a particular VR app is the best solution here. Something like a flag for `WebVRConfig.DISABLE_WAKE_LOCK = true` or the like.

**Comment by capnmidnight** [comment]
That sounds reasonable

**Comment by cvan** [comment]
Yikes, yeah, that's unfortunate.

ICYMI: there's a new [Wake Lock API](https://www.w3.org/TR/wake-lock/) being spec'd for requesting a wake lock.

Adding a pref in the meantime sounds like a decent solution. I wonder if there are in fact other wake-lock hacks; I haven't tested lately to see on which devices _not_ having wake-lock code becomes problematic.

**Comment by jsantell** [comment]
Fixed in 0.10.2 by adding `MOBILE_WAKE_LOCK` config; set to false to disable the WakeLock always
